src/RageSoundPosMap.cpp

```diff
@@ -54,54 +54,40 @@ pos_map_queue &pos_map_queue::operator=( const pos_map_queue &rhs )
 	return *this;
 }
 
-void pos_map_queue::Insert( std::int64_t iSourceFrame, int iFrames, std::int64_t iDestFrame, float fSourceToDestRatio )
+void pos_map_queue::Insert(std::int64_t iSourceFrame, std::int64_t iFrames, std::int64_t iDestFrame, double fSourceToDestRatio)
 {
-	if( !m_pImpl->m_Queue.empty() )
-	{
-		/* Optimization: If the last entry lines up with this new entry, just merge them. */
-		pos_map_t &last = m_pImpl->m_Queue.back();
-		if( last.m_iSourceFrame + last.m_iFrames == iSourceFrame &&
-		    last.m_fSourceToDestRatio == fSourceToDestRatio &&
-		    llabs(last.m_iDestFrame + std::lrint(last.m_iFrames * last.m_fSourceToDestRatio) - iDestFrame) <= 1 )
-		{
-			last.m_iFrames += iFrames;
-
-			/* Make sure that m_Frames doesn't grow too large and overflow an int. */
-			if( !m_pImpl->m_Queue.empty() && last.m_iFrames > pos_map_backlog_frames * 2 )
-			{
-				/*
-				 * Split this entry into two smaller entries.  This will cause up to one
-				 * sample of rounding error in m_iDestFrame.  This will not accumulate;
-				 * if we split again and it becomes two frames of error, the next Insert()
-				 * will be beyond the tolerance of the above iDestFrame check, and new
-				 * data will be added to a new entry.
-				 */
-				int iDeleteFrames = last.m_iFrames - pos_map_backlog_frames;
-
-				pos_map_t next(last);
-
-				last.m_iFrames = iDeleteFrames;
-
-				next.m_iSourceFrame += iDeleteFrames;
-				next.m_iFrames -= iDeleteFrames;
-				next.m_iDestFrame += std::lrint( iDeleteFrames * next.m_fSourceToDestRatio );
+	bool merged = false;
 
-				m_pImpl->m_Queue.push_back( next );
-			}
+	// Check if the is queue is empty or not
+	if (!m_pImpl->m_Queue.empty())
+	{
+		// Check if the last entry can be merged with the new entry
+		pos_map_t& last = m_pImpl->m_Queue.back();
+		if (last.m_iSourceFrame + last.m_iFrames == iSourceFrame &&
+			last.m_fSourceToDestRatio == fSourceToDestRatio &&
 
-			m_pImpl->Cleanup();
+			// llabs() is used instead of abs() because abs() would be susceptible to an integer overflow.
+			llabs(last.m_iDestFrame + static_cast<int64_t>(last.m_iFrames * last.m_fSourceToDestRatio + 0.5) - iDestFrame) <= 1)
 
-			return;
+		{
+			// Merge the frames and set the merged flag to true; and only run Cleanup() if the merged flag is false.
```

Comment:
We're not doing what this comment is saying... `Cleanup()` is being called all the time and not only when `merged` is false.

Why are we changing this behavior?

Reply:
> We're not doing what this comment is saying... `Cleanup()` is being called all the time and not only when `merged` is false.
> 
> Why are we changing this behavior?

Sorry for the confusion here. [Cleanup() was being called twice in the original code](https://github.com/itgmania/itgmania/blob/04647648765b082414993e6f2ce420b2d950840b/src/RageSoundPosMap.cpp#L92), so this is explaining that it's not being called twice if doesn't have to be. In the original code, this additional Cleanup() happens after the buffer is trimmed. The buffer trimming code is not present in this commit. I haven't seen any performance issues or detriment to sync stability by calling `Cleanup()` once instead of twice.

I see how the comment is misleading though, so I'll revise it.